File: app/src/main/java/com/acme/realmudi/model/TodoItem.java
package com.acme.realmudi.model;

import java.util.Date;
import java.util.UUID;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

public class TodoItem extends RealmObject {

    @PrimaryKey
    @Required
    private String id;

    @Required
    private Date createdDate;

    @Required
    private String text;

    private boolean selected;

    public TodoItem(final String text) {
        this.id = UUID.randomUUID().toString();
        this.createdDate = new Date();
        this.text = text;
    }

    // Getters / Setters

    public TodoItem() {}

    public String getId() {
        return id;
    }

    public String getText() {
        return text;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }
}

File: app/src/main/java/com/acme/realmudi/model/TodoItem.kt
package com.acme.realmudi.model

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import java.util.*

open class TodoItem : RealmObject {

    @PrimaryKey
    @Required
    var id: String? = null

    @Required
    private var createdDate: Date? = null

    @Required
    var text: String? = null

    var isSelected: Boolean = false

    constructor(text: String) {
        this.id = UUID.randomUUID().toString()
        this.createdDate = Date()
        this.text = text
    }

    constructor() {}

}